src/com/maddyhome/idea/vim/ex/handler/RegistersHandler.kt

```diff
@@ -21,15 +21,25 @@ package com.maddyhome.idea.vim.ex.handler
 import com.intellij.openapi.actionSystem.DataContext
 import com.intellij.openapi.editor.Editor
 import com.maddyhome.idea.vim.VimPlugin
+import com.maddyhome.idea.vim.command.SelectionType
 import com.maddyhome.idea.vim.ex.*
 import com.maddyhome.idea.vim.helper.StringHelper
 
 class RegistersHandler : CommandHandler.SingleExecution() {
   override val argFlags = flags(RangeFlag.RANGE_OPTIONAL, ArgumentFlag.ARGUMENT_OPTIONAL, Access.READ_ONLY)
   override fun execute(editor: Editor, context: DataContext, cmd: ExCommand): Boolean {
-    val regs = VimPlugin.getRegister().registers.joinToString("\n", prefix = "--- Registers ---\n") { reg ->
-      """"${reg.name}   ${StringHelper.toKeyNotation(reg.keys)}"""
-    }
+
+    val registerGroup = VimPlugin.getRegister()
+    val regs = registerGroup.registers
+      .filter { cmd.argument.isEmpty() || cmd.argument.contains(it.name) }
+      .joinToString("\n", prefix = "Type Name Content\n") { reg ->
```

Comment:
Why have you changed the output template? I mean, I think your version is better, but it differs from vim, isn't it?

Reply:
Ah, I changed it to match Vim on my machine, and it looks like Vim's output was [updated in 8.1.2212](https://github.com/vim/vim/commit/3691f1ee72b68a47e6dcc75927bfa46565cf3614#diff-19aa2e6e5c0b4e89aeee2e37be194690)

Reply:
Oh, I have an 8.1 version, but seems like an older update. Sorry for troubles, I didn't mean to send you to vim sources :)